lib/phoenix_live_view/channel.ex

```diff
@@ -629,8 +629,9 @@ defmodule Phoenix.LiveView.Channel do
     Process.monitor(transport_pid)
     load_csrf_token(endpoint, socket_session)
 
+    url = Map.fetch!(params, "url")
+    host = if url, do: URI.parse(url).host, else: "www.example.com" # for live_isolated
```

Comment:
I didn't make it an option on `live_isolated` because I wanted to avoid supporting an "invalid" client option at this stage

Reply:
We don't need to get the host here. Line 654 returns the parsed_uri and we can get the host at that moment, to avoid parsing it twice.

Reply:
It feels like testing concerns are leaking to the code though. For example, we can have a bug on the client which sends a URL without the domain and now everything is www.example.com. Can we enforce live_isolated to always pass the URL?

Reply:
Good call!